examples/tutorial/visuals/01_basic_visual.py

```diff
@@ -0,0 +1,165 @@
+vispy's base library of visuals will be sufficient to create complex scenes as
+needed. However, there are cases where a particular visual effect is desired 
+that is not supported in the base library, or when a custom visual is needed to
+optimize performance for a specific use case.
+
+The purpose of a Visual is to encapsulate a single drawable object. This
+drawable can be as simple or complex as desired. Some of the simplest visuals 
+draw points, lines, or triangles, whereas more complex visuals invove multiple
+drawing stages or make use of sub-visuals to construct larger objects.
+
+In this example we will create a very simple Visual that draws a rectangle.
+Visuals are defined by:
+
+1. Creating a subclass of vispy.visuals.Visual
+2. Defining a draw() method that takes into account some user-specified
+   transformation functions.
+
+
+"""
+from vispy import app, gloo, visuals, scene
+import numpy as np
+
+# Define a simple vertex shader. We use $template variables as placeholders for
+# code that will be inserted later on. In this example, $position will become
+# an attribute, and $transform will become a function. Important: using
+# $transform in this way ensures that users of this visual will be able to 
+# apply arbitrary transformations to it.
+vertex_shader = """
+void main() {
+   gl_Position = $transform(vec4($position, 0, 1));
+}
+"""
+
+# Very simple fragment shader. Again we use a template variable "$color", which
+# allows us to decide later how the color should be defined (in this case, we
+# will just use a uniform red color).
+fragment_shader = """
+void main() {
+  gl_FragColor = $color;
+}
+"""
+
+
+# Start the new Visual class. 
+# By convention, all Visual subclass names and in 'Visual'
+class MyRectVisual(visuals.Visual):
+    """Visual that draws a red rectangle.
+    
+    Parameters
+    ----------
+    x : float
+        x coordinate of rectangle origin
+    y : float
+        y coordinate of rectangle origin
+    w : float
+        width of rectangle
+    h : float
+        height of rectangle
+        
+    All parameters are specified in the local (arbitrary) coordinate system of
+    the visual. How this coordinate system translates to the canvas will 
+    depend on the transformation functions used during drawing.
+    """
+    
+    # There are no constraints on the signature of the __init__ method; use
+    # whatever makes the most sense for your visual.
+    def __init__(self, x, y, w, h):
+        visuals.Visual.__init__(self)
+        
+        # vertices for two triangles forming a rectangle
+        self.vbo = gloo.VertexBuffer(np.array([
+            [x, y], [x+w, y], [x+w, y+h],
+            [x, y], [x+w, y+h], [x, y+h]
+        ], dtype=np.float32))
+        
+        # We use a ModularProgram because it allows us to plug in arbitrary 
+        # transformation functions. This is recommended but not strictly 
+        # required.
```

Comment:
Don't we actually require it? Or do we check whether or not e.g. `SceneCanvas` has a place to put its transform?